runtime/lua/vim/diagnostic.lua

```diff
@@ -347,17 +347,13 @@ local function schedule_display(namespace, bufnr, args)
 
   local key = make_augroup_key(namespace, bufnr)
   if not registered_autocmds[key] then
-    vim.cmd(string.format(
-      [[augroup %s
-      au!
```

Comment:
your refactor is missing the `au!`, i.e., clearing previous autocommands in the augroup. You need to clear them manually.

Reply:
use `nvim_del_autocmd`  to clear prev autocmd?

Reply:
can't you just create an augroup with `clear = true` (which should be default iirc)

Reply:
use `nvim_clear_autocmds()` (see the similar refactor at https://github.com/nvim-treesitter/nvim-treesitter/commit/07eb437bb1c4c0a0298553dd509294f82362d6bc)

Reply:
yup check doc I found `clear = true`